### PR TITLE
Trim leading/trailing whitespace before parsing iri

### PR DESCRIPTION
**Pull request by eg-ryte**

### CODE REVIEW

File: src/test/scala/fr/davit/capturl/IriSpec.scala [hunk 1]
```diff
@@ -153,4 +153,9 @@ class IriSpec extends FlatSpec with Matchers {
     Iri("path?query#", Lazy) shouldBe Iri("path?query#", Strict)
   }
 
+  "Iri" should "ignore leading and trailing whitespace" in {
+    Iri(" http://example.com/path\t", Lazy) shouldBe Iri("http://example.com/path")
+    Iri(" http://example.com/path\t", Strict) shouldBe Iri("http://example.com/path")
```
Comment (sflip) [inline src/test/scala/fr/davit/capturl/IriSpec.scala]: A few more test cases with leading (or trailing respectively) whitespace only won't hurt.

Comment (eg-ryte) [reply src/test/scala/fr/davit/capturl/IriSpec.scala]: Wouldn't that essentially just test the `trim` method though?

Comment (sflip) [reply src/test/scala/fr/davit/capturl/IriSpec.scala]: Using `trim` is implementation, the test does and sould not not know this. (A bad implementation could check whether the string starts with whitespace and only trim in this case, e.g.)

**Comment by sflip** [comment]
@RustedBones! <3
